contracts/src/Creator.sol

```diff
@@ -13,17 +42,141 @@ contract Creator {
     // Storage
     // -----------------------------------------
 
+    /// @dev Streamers (that are a)
+    mapping (address => uint) public streamers;
+
     // -----------------------------------------
     // Constructor
     // -----------------------------------------
 
     // function postUpgrade(uint256 id) public proxied {}
 
-    constructor() {
-        
+    constructor() public ERC1155PresetMinterPauser() {
+        _mint(msg.sender, "{'name': 'test', 'description': 'testdesc', 'content': 'testurl'}", 1, ""); //1 means NFT.
+        ISuperfluid host,
+        IConstantFlowAgreementV1 cfa,
+        ISuperToken acceptedToken) {
+        assert(address(host) != address(0));
+        assert(address(cfa) != address(0));
+        assert(address(acceptedToken) != address(0));
+
+        _host = host;
+        _cfa = cfa;
+        _acceptedToken = acceptedToken;
+
+        uint256 configWord =
+            SuperAppDefinitions.TYPE_APP_FINAL;
+
+        _host.registerApp(configWord);
     }
 
     // -----------------------------------------
     // External Functions
     // -----------------------------------------
-}
\ No newline at end of file
+
+    /// @dev Take collateral fee from the user and add them acceptable streamers
```

Comment:
are 'user' and 'streamer' the same thing? maybe we should use one term so we don't confuse ourselves :) 
...I think subscriber or supporter would be the most clear